test/eigen.jl

```diff
@@ -43,6 +44,10 @@
         @test vals::SVector ≈ vals_a
         @test eigvals(m) ≈ vals
         @test (vecs*diagm(vals)*vecs')::SMatrix ≈ m
+        
+        m_d = randn(3)
+        m = @SMatrix diagm(m_d)
```

Comment:
Sorry, this isn't supported by the macro. I think `diagm(m_d)` works fine, however.